lib/ansible/modules/cloud/amazon/s3_cors.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: s3_cors
+short_description: Manage CORS for S3 buckets in AWS
+description:
+    - Manage CORS for S3 buckets in AWS
+version_added: "2.4"
+author: "Oyvind Saltvik (@fivethreeo)"
+options:
+  name:
+    description:
+      - Name of the s3 bucket
+    required: true
+    default: null
+  rules:
+    description:
+      - Cors rules to put on the s3 bucket
+    required: false
+  state:
+    description:
+      - Create or remove cors on the s3 bucket
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a simple cors for s3 bucket
+- s3_cors:
+    name: mys3bucket
+    rules:
+      - allowed_origins:
+          - http://www.example.com/
+        allowed_methods:
+          - GET
+          - POST
+        allowed_headers:
+          - Authorization
+        expose_headers:
+          - x-amz-server-side-encryption
+          - x-amz-request-id
+        max_age_seconds: 30000
+
+# Remove cors for s3 bucket
+- s3_cors:
+    name: mys3bucket
+    state: absent
+'''
+
+RETURN = '''
+changed:
+  description: check to see if a change was made to the rules
+  returned: always
+  type: boolean
+  sample: true
+name:
+  description: name of bucket
+  returned: always
+  type: string
+  sample: 'bucket-name'
+rules:
+  description: list of current rules
+  returned: always
+  type: list
+  sample: [
+     {
+        "allowed_headers": [
+          "Authorization"
+        ],
+        "allowed_methods": [
+          "GET"
+        ],
+        "allowed_origins": [
+          "*"
+        ],
+        "max_age_seconds": 30000
+      }
+    ]
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (HAS_BOTO3, boto3_conn, ec2_argument_spec, get_aws_connection_info,
+                                      camel_dict_to_snake_dict, _botocore_exception_maybe)
+
+ClientError = _botocore_exception_maybe()
+
+
+def snake_dict_to_camel_dict(snake_dict):
```

Comment:
Pain of nearly duplicate code. :( I'm not sure what the best approach is for this. Here is an example of how this was dealt with elsewhere recently: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/elb_application_lb.py#L148-L156
Requiring the user to specify the correct casing. What do you think about that approach? And maybe what you're doing here is fine (though it seems like this problem may be recurring, in which case making a module_utils function that is snake_dict_to_camel_dict but with capitalized first words may be a solution.).

@ryansb Thoughts on this?